sdk/tracing/src/main/java/io/opentelemetry/sdk/trace/export/BatchSpanProcessor.java

```diff
@@ -302,10 +296,19 @@ public void run() {
         } catch (Exception e) {
           logger.log(Level.WARNING, "Exporter threw an Exception", e);
         }
+        return exportSize;
       } else {
-        logger.log(Level.FINE, "Exporter busy. Dropping spans.");
+        return 0;
       }
     }
+
+    private static void dropSpans(ArrayList<ReadableSpan> spansList, int count) {
+      int remaining = spansList.size() - count;
+      ReadableSpan[] retainedSpanList = new ReadableSpan[remaining];
+      System.arraycopy(spansList.toArray(), count, retainedSpanList, 0, remaining);
+      spansList.clear();
+      spansList.addAll(Arrays.asList(retainedSpanList));
+    }
```

Comment:
There has to be a better way of dropping elements from an array... Given that `clear` traverses the array and nulls elements, we've effectively got three traverses of the array. Now in practical terms, `retainedSpanList ` should be empty most times, or at least only a few elements. So the array copying at L308 and L310 is going to be negligible.